File: 1_java_delombok/books/src/main/java/com/devtiro/database/domain/dto/BookDto.java
package com.devtiro.database.domain.dto;

public class BookDto {

    private String isbn;

    private String title;

    private AuthorDto author;

    public BookDto(String isbn, String title, AuthorDto author) {
        this.isbn = isbn;
        this.title = title;
        this.author = author;
    }

    public BookDto() {
    }

    public static BookDtoBuilder builder() {
        return new BookDtoBuilder();
    }

    public String getIsbn() {
        return this.isbn;
    }

    public String getTitle() {
        return this.title;
    }

    public AuthorDto getAuthor() {
        return this.author;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setAuthor(AuthorDto author) {
        this.author = author;
    }

    public boolean equals(final Object o) {
        if (o == this) return true;
        if (!(o instanceof BookDto)) return false;
        final BookDto other = (BookDto) o;
        if (!other.canEqual((Object) this)) return false;
        final Object this$isbn = this.getIsbn();
        final Object other$isbn = other.getIsbn();
        if (this$isbn == null ? other$isbn != null : !this$isbn.equals(other$isbn)) return false;
        final Object this$title = this.getTitle();
        final Object other$title = other.getTitle();
        if (this$title == null ? other$title != null : !this$title.equals(other$title)) return false;
        final Object this$author = this.getAuthor();
        final Object other$author = other.getAuthor();
        if (this$author == null ? other$author != null : !this$author.equals(other$author)) return false;
        return true;
    }

    protected boolean canEqual(final Object other) {
        return other instanceof BookDto;
    }

    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        final Object $isbn = this.getIsbn();
        result = result * PRIME + ($isbn == null ? 43 : $isbn.hashCode());
        final Object $title = this.getTitle();
        result = result * PRIME + ($title == null ? 43 : $title.hashCode());
        final Object $author = this.getAuthor();
        result = result * PRIME + ($author == null ? 43 : $author.hashCode());
        return result;
    }

    public String toString() {
        return "BookDto(isbn=" + this.getIsbn() + ", title=" + this.getTitle() + ", author=" + this.getAuthor() + ")";
    }

    public static class BookDtoBuilder {
        private String isbn;
        private String title;
        private AuthorDto author;

        BookDtoBuilder() {
        }

        public BookDtoBuilder isbn(String isbn) {
            this.isbn = isbn;
            return this;
        }

        public BookDtoBuilder title(String title) {
            this.title = title;
            return this;
        }

        public BookDtoBuilder author(AuthorDto author) {
            this.author = author;
            return this;
        }

        public BookDto build() {
            return new BookDto(this.isbn, this.title, this.author);
        }

        public String toString() {
            return "BookDto.BookDtoBuilder(isbn=" + this.isbn + ", title=" + this.title + ", author=" + this.author + ")";
        }
    }
}

File: 2_kotlin/books/src/main/kotlin/com/devtiro/database/domain/dto/BookDto.kt
package com.devtiro.database.domain.dto

class BookDto {
    @JvmField
    var isbn: String? = null

    @JvmField
    var title: String? = null

    var author: AuthorDto? = null

    constructor(isbn: String?, title: String?, author: AuthorDto?) {
        this.isbn = isbn
        this.title = title
        this.author = author
    }

    constructor()

    override fun equals(o: Any?): Boolean {
        if (o === this) return true
        if (o !is BookDto) return false
        val other = o
        if (!other.canEqual(this as Any)) return false
        val `this$isbn`: Any? = this.isbn
        val `other$isbn`: Any? = other.isbn
        if (if (`this$isbn` == null) `other$isbn` != null else `this$isbn` != `other$isbn`) return false
        val `this$title`: Any? = this.title
        val `other$title`: Any? = other.title
        if (if (`this$title` == null) `other$title` != null else `this$title` != `other$title`) return false
        val `this$author`: Any? = this.author
        val `other$author`: Any? = other.author
        if (if (`this$author` == null) `other$author` != null else `this$author` != `other$author`) return false
        return true
    }

    protected fun canEqual(other: Any?): Boolean {
        return other is BookDto
    }

    override fun hashCode(): Int {
        val PRIME = 59
        var result = 1
        val `$isbn`: Any? = this.isbn
        result = result * PRIME + (`$isbn`?.hashCode() ?: 43)
        val `$title`: Any? = this.title
        result = result * PRIME + (`$title`?.hashCode() ?: 43)
        val `$author`: Any? = this.author
        result = result * PRIME + (`$author`?.hashCode() ?: 43)
        return result
    }

    override fun toString(): String {
        return "BookDto(isbn=" + this.isbn + ", title=" + this.title + ", author=" + this.author + ")"
    }

    class BookDtoBuilder internal constructor() {
        private var isbn: String? = null
        private var title: String? = null
        private var author: AuthorDto? = null
        fun isbn(isbn: String?): BookDtoBuilder {
            this.isbn = isbn
            return this
        }

        fun title(title: String?): BookDtoBuilder {
            this.title = title
            return this
        }

        fun author(author: AuthorDto?): BookDtoBuilder {
            this.author = author
            return this
        }

        fun build(): BookDto {
            return BookDto(this.isbn, this.title, this.author)
        }

        override fun toString(): String {
            return "BookDto.BookDtoBuilder(isbn=" + this.isbn + ", title=" + this.title + ", author=" + this.author + ")"
        }
    }

    companion object {
        @JvmStatic
        fun builder(): BookDtoBuilder {
            return BookDtoBuilder()
        }
    }
}
